ml-proto/src/script.ml

```diff
@@ -71,6 +75,42 @@ let run_command cmd =
       Error.error cmd.at "assertion failed"
     end
 
+  | AssertHeapEq (offset, expected) ->
+    let m = match !current_module with
+      | Some m -> m
+      | None -> Error.error cmd.at "no module defined to examine heap"
+    in
+    let mem = Eval.get_module_memory m in
+    let count = String.length expected in
+
+    for i=0 to (count-1) do
+      let expected_ch = Char.code expected.[i] in
+      let actual_ch = match Memory.load mem i UInt8Mem Int32Type with
+        | Int32 i -> Int32.to_int i
+        | _ -> Error.error cmd.at "memory load returned wrong type"
```

Comment:
This would be an internal error in (the memory module of) the interpreter, not an error in the program, so the `Error` exception seems inappropriate. One idiomatic way of indicating such (hopefully) unreachable code in Ocaml is `assert false`.